config/config-api/src/com/thoughtworks/go/config/materials/PluggableSCMMaterialConfig.java

```diff
@@ -315,11 +315,11 @@ public boolean equals(Object o) {
             return false;
         }
 
-        if (scmConfig == null || that.scmConfig == null) {
+        if (scmConfig != null ? !scmConfig.equals(that.scmConfig) : that.scmConfig != null) {
             return false;
         }
 
-        return this.getFingerprint().equals(that.getFingerprint());
```

Comment:
fingerprint is a computed field which internally checks for scmConfig itself. If you see, I have included scmConfig field for equals comparison and removed fingerprint, as is done for the other materials. Also, while saving a config xml, the scmConfig object could be null, and only the ref/scmId would be set. In such a case, equals comparison returned false for materials pointing to the same ref/scmId